webapi/src/main/scala/org/knora/webapi/responders/v1/ResourcesResponderV1.scala

```diff
@@ -74,244 +72,291 @@ class ResourcesResponderV1 extends ResponderV1 {
-                case None =>
-                    GraphNodeV1(id = resIri,
-                        resinfo = None,
-                        properties = None)
-            }
-        }
+                response: SparqlSelectResponse <- (storeManager ? SparqlSelectRequest(sparql)).mapTo[SparqlSelectResponse]
+                rows = response.results.bindings
+
+                // Did we get any results?
+                recursiveResults: GraphQueryResults <- if (rows.isEmpty) {
+                    // No. Return  nothing.
+                    Future(GraphQueryResults())
+                } else {
+                    // Yes. Get the nodes from the query results.
+                        val otherNodes: Seq[QueryResultNode] = rows.map {
+                            row =>
+                                val rowMap = row.rowMap
+
+                                QueryResultNode(
+                                    nodeIri = rowMap("node"),
+                                    nodeClass = rowMap("nodeClass"),
+                                    nodeLabel = rowMap("nodeLabel"),
+                                    nodeOwner = rowMap("nodeOwner"),
+                                    nodeProject = rowMap("nodeProject"),
+                                    nodePermissions = rowMap.get("nodePermissions")
+                                )
+                        }.filter {
+                            node =>
+                                // Filter out the nodes that the user doesn't have permission to see.
+                                PermissionUtilV1.getUserPermissionV1(
+                                    subjectIri = node.nodeIri,
+                                    subjectOwner = node.nodeOwner,
+                                    subjectProject = node.nodeProject,
+                                    subjectPermissionLiteral = node.nodePermissions,
+                                    userProfile = graphDataGetRequest.userProfile
+                                ).nonEmpty
+                        }
 
-        // TODO: filter resinfo and properties for literal V1 frontend compatibility (100% necessary?)
-        // TODO: switch to twirl template
+                        // Collect the IRIs of the nodes that the user has permission to see, including the start node.
+                        val visibleNodeIris = otherNodes.map(_.nodeIri).toSet + startNode.nodeIri
+
+                        // Get the edges from the query results.
+                        val edges: Seq[QueryResultEdge] = rows.map {
+                            row =>
+                                val rowMap = row.rowMap
+                                val nodeIri = rowMap("node")
+
+                                QueryResultEdge(
+                                    linkValueIri = rowMap("linkValue"),
+                                    sourceNodeIri = if (outbound) startNode.nodeIri else nodeIri,
```

Comment:
Could you add a comment here explaining why the Iris are swapped depending on the state of outbound (outgoing or incoming links)?